Cargo.toml

```diff
@@ -42,6 +42,7 @@ thiserror = "1"
 tokio = "1"
 tracing = "0.1"
 tracing-subscriber = "0.3"
+ttl_cache = "0.5"
```

Comment:
We already depend on the `cached` crate. Is there something that crate cannot offer compared to `ttl_cache`?

Reply:
Nice, I thought there was something like that already in the codebase. One slight annoyance with cached's `[TimedCache](https://docs.rs/cached/0.49.2/cached/stores/struct.TimedCache.html#)` is that even reads take a &mut :-/. It does give non mutable access to the store but then we have to do the time checking logic ourselves.